vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc

```diff
@@ -226,10 +226,14 @@ void LedgerImpl::OnLedgerStateLoaded(ledger::Result result,
       OnWalletInitialized(ledger::Result::INVALID_LEDGER_STATE);
     } else {
       auto wallet_info = bat_state_->GetWalletInfo();
-      SetConfirmationsWalletInfo(wallet_info);
-
-      LoadPublisherState(this);
-      bat_contribution_->OnStartUp();
+      if (GetPaymentId().empty() || GetWalletPassphrase().empty()) {
+        OnWalletInitialized(ledger::Result::CORRUPTED_WALLET);
+      } else {
+        SetConfirmationsWalletInfo(wallet_info);
+
+        LoadPublisherState(this);
```

Comment:
even if wallet is corrupted, we need to load other stuff

Reply:
In this particular case where panel and page won't interact with that data, we should still load? This section will still run if they create a new wallet on corrupt.

Reply:
if seed is corrupted for example user can still load everything and do recovery on rewards page. If we block it then user will be in strange state

Reply:
problem is that you now return two `OnWalletInitialized` calls if we load other stuff

Reply:
I moved check to after publisher_state load